### ISSUE TITLE
Failed to establish the environment: The detected CUDA version (10.1) mismatches the version that was used to compile PyTorch (11.7). Please make sure to use the same CUDA versions.

### ISSUE THREAD

**Issue by Sissi-lu** [aeolusguan/NMRF#7]
Hi, thanks for your wonderful work!

When I establish the environment for running the ops/make.sh. A error occurs as follows:

The detected CUDA version (10.1) mismatches the version that was used to compile
PyTorch (11.7). Please make sure to use the same CUDA versions.

However, I have installed the exact pytorch==1.13 and cuda version==11.7.
![image](https://github.com/user-attachments/assets/6d392154-3525-45d6-b1f3-892a38d25b3f)

How can I solve this problem?

Thanks in advance!

**Comment by aeolusguan** [comment]
please check the cuda version under the directory `from torch.utils import cpp_extension; print(cpp_extension.CUDA_HOME)`.

**Comment by Sissi-lu** [comment]
Thank you for your quick response. I checked in this way and found out it is ``usr/local/cuda-10.1``.
So what can I do next？
Thanks again.

**Comment by aeolusguan** [comment]
A simple workaround maybe creating a symbolic link to your install coda-11.7, i.e., `cd /usr/local; sudo ln -s /your/path/to/cuda-11.7 cuda`.

**Comment by Sissi-lu** [comment]
Hi, thank you for your response. I cannot use the sudo order, so I found another way to solve this problem. Write in here to remind me:

0. conda activate NMRF
1. select the right cuda-nvcc version in https://anaconda.org/nvidia/repo, e.g.,
`conda install nvidia/label/cuda-11.7.0::cuda-nvcc`
2. select the right cuda-libraries-dev in https://anaconda.org/nvidia/cuda-libraries-dev, e.g., and install with conda
`conda install nvidia/label/cuda-11.7.0::cuda-libraries-dev
`
3. then, running this:
4. `export CUDA_HOME=/your/environment/path`
5. run the setup.py to see if there is any wrong
6. according to the specific error, set the symbolic link to the environment, .e.g.,
`export LD_LIBRARES_PATH=/xxxx/.conda/envs/NMRF/lib
ls -la /xxx/.conda/envs/NMRF/lib/libcudart.so
`

Thank you again for helping me identify the difference between `torch.version.cuda` and `cpp_extension.CUDA_HOME`.
